lcd.lua

```diff
@@ -165,30 +198,28 @@ minetest.register_node("digilines:lcd", {
 	inventory_image = "lcd_lcd.png",
 	wield_image = "lcd_lcd.png",
 	tiles = {"lcd_anyside.png"},
-
 	paramtype = "light",
 	sunlight_propagates = true,
+	light_source = 6,
 	paramtype2 = "wallmounted",
 	node_box = lcd_box,
 	selection_box = lcd_box,
 	groups = {choppy = 3, dig_immediate = 2},
-
-	after_place_node = function (pos)
+	after_place_node = function(pos)
 		local param2 = minetest.get_node(pos).param2
 		if param2 == 0 or param2 == 1 then
 			minetest.add_node(pos, {name = "digilines:lcd", param2 = 3})
 		end
-		prepare_writing (pos)
-	end,
-
-	on_construct = function(pos)
-		reset_meta(pos)
+		spawn_entity(pos)
+		prepare_writing(pos)
 	end,
-
-	on_destruct = function(pos)
-		clearscreen(pos)
+	on_construct = reset_meta,
+	on_destruct = clearscreen,
+	on_punch = function(pos, node, puncher, pointed_thing)
+		if puncher:is_player() and minetest.get_player_by_name(puncher:get_player_name()) then
+			spawn_entity(pos)
+		end
```

Comment:
Because some punchers aren't players

Reply:
If you were to have something punch the LCD repeatedly whilst in a sub-loaded chunk, you now have the same problem we are trying to fix.

Reply:
I'd recommend using `minetest.is_player(puncher) ` because it includes the `puncher:is_player()` check. Offline players are not valid players.

Also `get_player_name()` returns `""` for non-players. https://github.com/minetest/minetest/blob/dd5f03731ad545f52da24825a2ee870f899f6adc/doc/lua_api.txt#L3325

Reply:
Then shouldn't I just use `if puncher:get_player_name() ~= ""`?

Reply:
That is possible too. The advantage of `minetest.is_player` is that it accepts all values and is guaranteed to return a boolean value indicating whether it's a player (or not).